src/components/Admin/Authentication/Util.tsx

```diff
@@ -4,9 +4,147 @@ import {
   FormControl,
   FormErrorMessage,
   CircularProgress,
+  InputRightElement,
+  FormLabel,
+  Input,
+  InputGroup,
 } from '@chakra-ui/react';
 import firebase, {firebaseAuth} from '../../../firebase';
 
+/**
+ * Props for PasswordVisibilityToggle component. Used for type safety.
+ */
+interface PasswordVisibilityProps {
+  showPassword: boolean;
+  handlePasswordVisibility: () => void;
+}
+
+/**
+ * Component for inside password input fields that toggles if a password is
+ * visible in text or hidden behind dots.
+ * @param props - showPassword, handlePasswordVisibility
+ * - `showPassword` whether or not password is hidden
+ * - `handlePasswordVisibility` changes showPassword
+ */
+const PasswordVisibilityToggle: ({
+  showPassword,
+  handlePasswordVisibility,
+}: PasswordVisibilityProps) => JSX.Element = ({
+  showPassword,
+  handlePasswordVisibility,
+}: PasswordVisibilityProps) => {
+  return (
+    <InputRightElement width="4.5rem">
+      <Button
+        height="1.5rem"
+        size="sm"
+        verticalAlign="center"
```

Comment:
It looks like it's not perfectly center (shifted towards the top) in the actual form. I don't think it needs to change, just thought it was worth pointing out.

Reply:
Yeah this was weird, but I think it looks better after the changes I just added. Adjusting the Input "size" fixed it. I think it's because the "size" part of the Input and the Button of the show/hide password button determine the vertical placement, not the verticalAlign feature, but I'm not completely sure.